templates/autosd-app-template/template.yaml

```
@@ -61,23 +61,39 @@ spec:
               default: latest
               description: Build Image tag
 
-    - title: Provide Cluster Information
-      required:
-        - guid
-        - base_domain
-      properties:
-        guid:
-          title: Cluster GUID
-          type: string
-          description: "Unique identifier of the cluster. GUID = 6xw6w in https://..cluster-6xw6w.6xw6w.sandbox4358.opentlc.com. See the URL above !"
-          maxLength: 5
-        base_domain:
-          title: Cluster Domain
-          type: string
-          default: sandbox4358.opentlc.com
-          description: The base domain the cluster is running in. See the URL above !
-
   steps:
+    - id: request-url
+      name: Get cluster URL
+      action: http:backstage:request
+      input:
+        method: GET
+        path: /catalog/entities/by-name/template/default/autosd-app-template
+        headers:
+          Accept-Encoding: identity
+
+    - id: extract-url
+      name: Extract cluster URL
+      action: roadiehq:utils:jsonata
+      input:
+        data: ${{ steps['request-url'].output.body }}
+        expression: metadata.annotations."backstage.io/source-location"
+
+    - id: parse-url
+      name: Parse cluster URL
+      action: regex:replace
+      input:
+        regExps:
+          - pattern: (?:.*cluster-)(\w{5})(?:.*)
+            replacement: $1
+            values:
+              - key: guid
+                value: ${{ steps['extract-url'].output.result }}
+          - pattern: (?:.*)(cluster-\w{5}.*\.com)(?:.*)
```

```suggestion
          - pattern: (?:.*)(cluster-\w{5}.*\.com)(?:.*)
```

My only question is.. if during the LAB, everything will still be running under a .com domain.

Let me check with them.

---

Ok acording to them, it will be .com for aws deployments, and that is ours..., so we're fine.